src/client/src/database.rs

```diff
@@ -113,15 +115,16 @@ impl Database {
             .await
             .map_err(|e| {
                 let code = get_metadata_value(&e, INNER_ERROR_CODE)
-                    .and_then(|s| s.parse::<u32>().ok())
-                    .unwrap_or(e.code() as u32);
-                let err_msg = get_metadata_value(&e, INNER_ERROR_MSG).unwrap_or(e.to_string());
-                error::FlightGetSnafu {
-                    addr: client.addr(),
-                    code,
-                    err_msg,
-                }
-                .build()
+                    .and_then(|s| StatusCode::from_str(&s).ok())
+                    .unwrap_or(StatusCode::Unknown);
```

Comment:
Tonic's error code (`e.code()`) is ignored if INNER_ERROR_CODE is not set.

Reply:
If we reach this step mean tonic rpc is success? Because we can read messages from server's flight metadata.

Reply:
not necessarily. better keep the tonic's code